hcl/hcl.go

```diff
@@ -85,11 +88,48 @@ const (
 	ErrStackInvalidRunOrder     errutil.Error = "invalid stack execution order definition"
 )
 
-func NewConfig(reqversion string) Config {
+// NewConfig creates a new HCL config with absdir as config directory path.
+func NewConfig(absdir string, reqversion string) (Config, error) {
+	st, err := os.Stat(absdir)
+	if err != nil {
+		return Config{}, fmt.Errorf("initializing config: %w", err)
+	}
+
+	if !st.IsDir() {
+		return Config{}, fmt.Errorf("config constructor requires a directory path")
+	}
+
+	var tmblock *Terramate
+	if reqversion != "" {
+		tmblock = NewTerramate(reqversion)
+	}
+
 	return Config{
-		Terramate: &Terramate{
-			RequiredVersion: reqversion,
-		},
+		absdir:    absdir,
+		Terramate: tmblock,
+	}, nil
+}
+
+// AbsDir returns the absolute path of the configuration directory.
+func (c Config) AbsDir() string { return c.absdir }
+
+// Save the configuration file using filename inside config directory.
+func (c Config) Save(filename string) error {
+	cfgpath := filepath.Join(c.absdir, filename)
+	f, err := os.Create(cfgpath)
+	if err != nil {
+		return fmt.Errorf("saving configuration file %q: %w", cfgpath, err)
+	}
+
+	defer f.Close()
```

Comment:
If the file fails to close, indicating that maybe not everything was written, it will fail silently (it is extremelly annoying to handle this on all Close calls...on services I used to have a closer that just logs errors and use it as a wrapper everywhere). Also I think linters can warn about this, golangci-lint default options are indeed way too nice =P.

Reply:
I think we have other closes like this, so maybe we can address this on the future. But it should be addressed someday (the linter would help catch all of them).

Reply:
yes this should be handled, I forgot to fix this.